[.Net 10] [Windows] Build failure when using Identity.Client.Desktop.WinUI3 nuget

### Description

Referencing Microsoft.Identity.Client.Desktop.WinUI3 in a project with .net10 target results in build error

Could not copy the file ".nuget\packages\microsoft.maui.controls.core\10.0.10\lib\net10.0-windows10.0.19041\Microsoft.Maui\Handlers\HybridWebView\HybridWebView.js"

### Steps to Reproduce

1. Create a new maui project targeting .Net 10
2. Add following nuget packages references.

```
<ItemGroup Condition="'$(TargetFramework)' == 'net10.0-windows10.0.19041.0'">
  <PackageReference Include="Microsoft.Identity.Client" Version="4.79.1" />
  <PackageReference Include="Microsoft.Identity.Client.Desktop.WinUI3" Version="4.79.1"/>
</ItemGroup>
```
3. Build the project.

Expected: Build successful.
Actual: Following errors

```
C:\Program Files\Microsoft Visual Studio\18\Community\MSBuild\Current\Bin\arm64\Microsoft.Common.CurrentVersion.targets(5381,5): error MSB3030: Could not copy the file "C:\Users\***\.nuget\packages\microsoft.maui.controls.core\10.0.10\lib\net10.0-windows10.0.19041\Microsoft.Maui\Handlers\HybridWebView\HybridWebView.js" because it was not found.
C:\Program Files\Microsoft Visual Studio\18\Community\MSBuild\Current\Bin\arm64\Microsoft.Common.CurrentVersion.targets(5381,5): error MSB3030: Could not copy the file "C:\Users\***\.nuget\packages\microsoft.maui.core\10.0.10\lib\net10.0-windows10.0.19041\Microsoft.Maui\Handlers\HybridWebView\HybridWebView.js" because it was not found.
```

### Link to public reproduction project repository

_No response_

### Version with bug

10.0.10

### Is this a regression from previous behavior?

Yes, this used to work in .NET MAUI

### Last version that worked well

9.0.120 SR12

### Affected platforms

Windows

### Affected platform versions

Windows 19041

### Did you find any workaround?

None.

### Relevant log output

```shell

```

--- a/Directory.Build.props
+++ b/Directory.Build.props
@@ -149,6 +149,15 @@
     <EnablePreviewMsixTooling>true</EnablePreviewMsixTooling>
   </PropertyGroup>
 
+  <!-- TypeScript compilation -->
+  <PropertyGroup>
+    <TypeScriptTarget>ES2020</TypeScriptTarget>
+    <TypeScriptSourceMap>false</TypeScriptSourceMap>
+    <TypeScriptNoImplicitAny>true</TypeScriptNoImplicitAny>
+    <TypeScriptCompileOnSaveEnabled>true</TypeScriptCompileOnSaveEnabled>
+    <TypeScriptCompileBlocked Condition=" '$(TypeScriptCompileBlocked)' == '' and '$(CI)' != 'true' and '$(TF_BUILD)' != 'true' ">true</TypeScriptCompileBlocked>
+  </PropertyGroup>
+
   <!-- Windows specific settings -->
   <PropertyGroup Condition="'$(_MauiTargetPlatformIsWindows)' == 'True'">
     <WindowsSdkPackageVersion>10.0.19041.44</WindowsSdkPackageVersion>

--- a/Directory.Build.targets
+++ b/Directory.Build.targets
@@ -122,5 +122,7 @@
 
   <!-- Tell typescript to stop deleting everything before building the next TFM -->
   <Target Name="TypeScriptDeleteOutputFromOtherConfigs" />
+  <!-- Don't include any JS files in the Content, we will embed it -->
+  <Target Name="GetTypeScriptOutputForPublishing" />
 
 </Project>

--- a/src/Core/src/Core.csproj
+++ b/src/Core/src/Core.csproj
@@ -77,13 +77,6 @@
   </ItemGroup>
 
   <!-- TypeScript compilation -->
-  <PropertyGroup>
-    <TypeScriptTarget>ES2020</TypeScriptTarget>
-    <TypeScriptSourceMap>false</TypeScriptSourceMap>
-    <TypeScriptNoImplicitAny>true</TypeScriptNoImplicitAny>
-    <TypeScriptCompileOnSaveEnabled>true</TypeScriptCompileOnSaveEnabled>
-    <TypeScriptCompileBlocked Condition=" '$(TypeScriptCompileBlocked)' == '' and '$(CI)' != 'true' and '$(TF_BUILD)' != 'true' ">true</TypeScriptCompileBlocked>
-  </PropertyGroup>
   <ItemGroup>
     <PackageReference Include="Microsoft.TypeScript.MSBuild" Version="5.7.1">
       <PrivateAssets>all</PrivateAssets>
@@ -92,6 +85,7 @@
   </ItemGroup>
   <ItemGroup>
     <EmbeddedResource Include="Handlers\HybridWebView\HybridWebView.js" LogicalName="_framework/hybridwebview.js" />
+    <None Remove="Handlers\HybridWebView\HybridWebView.js" />
   </ItemGroup>
 
   <Target Name="_CopyToBuildTasksDir" AfterTargets="Build">

--- a/src/TestUtils/src/Microsoft.Maui.IntegrationTests/WindowsTemplateTest.cs
+++ b/src/TestUtils/src/Microsoft.Maui.IntegrationTests/WindowsTemplateTest.cs
@@ -179,4 +179,27 @@ public class WindowsTemplateTest : BaseTemplateTests
 				$"Unable to find expected asset: {fullpath}");
 		}
 	}
+
+	[Test]
+	public void BuildWithIdentityClient()
+	{
+		var projectDir = TestDirectory;
+		var projectFile = Path.Combine(projectDir, $"{Path.GetFileName(projectDir)}.csproj");
+
+		Assert.IsTrue(DotnetInternal.New("maui", projectDir, DotNetCurrent),
+			$"Unable to create template maui. Check test output for errors.");
+
+		// .NET 9 and later was Unpackaged, so we need to remove the line
+		FileUtilities.ReplaceInFile(projectFile,
+			"</Project>",
+			"""
+			<ItemGroup Condition="$(TargetFramework.Contains('-windows'))">
+			  <PackageReference Include="Microsoft.Identity.Client" Version="4.79.1" />
+			  <PackageReference Include="Microsoft.Identity.Client.Desktop.WinUI3" Version="4.79.1"/>
+			</ItemGroup>
+			""");
+
+		Assert.IsTrue(DotnetInternal.Build(projectFile, "Debug", properties: BuildProps, msbuildWarningsAsErrors: true),
+			$"Project {Path.GetFileName(projectFile)} failed to build. Check test output/attachments for errors.");
+	}
 }

--- a/src/TestUtils/src/Microsoft.Maui.IntegrationTests/WindowsTemplateTest.cs
+++ b/src/TestUtils/src/Microsoft.Maui.IntegrationTests/WindowsTemplateTest.cs
@@ -197,6 +197,7 @@ public class WindowsTemplateTest : BaseTemplateTests
 			  <PackageReference Include="Microsoft.Identity.Client" Version="4.79.1" />
 			  <PackageReference Include="Microsoft.Identity.Client.Desktop.WinUI3" Version="4.79.1"/>
 			</ItemGroup>
+			</Project>
 			""");
 
 		Assert.IsTrue(DotnetInternal.Build(projectFile, "Debug", properties: BuildProps, msbuildWarningsAsErrors: true),

